src/Dapr.Client/DaprClient.cs

```diff
@@ -143,6 +143,16 @@ public abstract class DaprClient
         /// <returns>A <see cref="ValueTask{T}" /> that will return the value when the operation has completed.</returns>
         public abstract ValueTask<TValue> GetStateAsync<TValue>(string storeName, string key, ConsistencyMode? consistencyMode = default, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Gets a list of values associated with the <paramref name="keys" /> from the Dapr state store.
+        /// </summary>
+        /// <param name="storeName">The name of state store to read from.</param>
+        /// <param name="keys">The list of keys to get values for.</param>
+        /// <param name="parallelism">The number of concurrent get operations the Dapr runtime will issue to the state store.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <returns>A <see cref="ValueTask{GetBulkItem}" /> that will return the list of values when the operation has completed.</returns>
+        
+        public abstract ValueTask<IEnumerable<GetBulkItem>> GetBulkStateAsync(string storeName, IEnumerable<string> keys, int parallelism = -1, CancellationToken cancellationToken = default);
```

Comment:
suggestion: use `int?` instead of a random sentinel value

Reply:
suggestion: return `ValueTask<IReadOnlyList<GetBulkItem>>` - this will allow getting the count, indexing, and iterating without an allocation.

Since the results don't truly stream from the server there's no advantage to `IEnumerable<>`

Reply:
> suggestion: return `ValueTask<IReadOnlyList<GetBulkItem>>` - this will allow getting the count, indexing, and iterating without an allocation.
> 
> Since the results don't truly stream from the server there's no advantage to `IEnumerable<>`

Changed to `IReadOnlyList<>`

Reply:
> suggestion: use `int?` instead of a random sentinel value

This isn't a random value and the param should be optional. Updated the comment.

Reply:
Yeah I get it - it's a little bit non-idiomatic to see sentinel values as part of an API when we have a whole language feature for that in C#. 

There's a very subtle way in which using `null` for "unspecified" is better than using `-1` (which I assume is what Dapr understands in the gRPC interface).  Default argument values get compiled **into** consumers of the library. So if you ever wanted to change the default here, it would not take effect for someone until they recompiled. 

Feel free to ignore if this seems unimportant.